src/main/java/chess/controller/ChessGameController.java

```diff
@@ -3,64 +3,71 @@
 import chess.domain.chessboard.ChessBoard;
 import chess.domain.chessboard.Square;
 import chess.domain.chessgame.ChessGame;
-import chess.domain.chessgame.GameCommand;
-import chess.dto.ChessBoardDto;
+import chess.domain.chessgame.gamecommand.EndState;
+import chess.domain.chessgame.gamecommand.GameCommandState;
+import chess.domain.chessgame.gamecommand.GameElements;
+import chess.domain.chesspiece.Camp;
+import chess.service.ChessGameService;
 import chess.view.InputView;
 import chess.view.OutputView;
 import java.util.List;
 import java.util.function.Supplier;
 
 public class ChessGameController {
 
-    private static final Integer GAME_COMMAND_INDEX = 0;
+    private final ChessGameService chessGameService = new ChessGameService();
 
     public void run() {
         OutputView.printStartMessage();
         gameStart();
     }
 
     private void gameStart() {
-        List<String> input = repeatUntilSuccess(InputView::requestGameCommand);
-        String inputCommand = input.get(GAME_COMMAND_INDEX);
-        GameCommand gameCommand = GameCommand.findGameCommand(inputCommand);
         try {
-            executeGameUntilEnd(gameCommand, input);
+            chessGameService.initialSetting();
+            ChessBoard chessBoard = chessGameService.chessBoardSetting();
+            executeGameCommand(chessBoard);
         } catch (IllegalArgumentException e) {
             OutputView.printErrorMessage(e.getMessage());
             gameStart();
         }
     }
 
-    private void executeGameUntilEnd(GameCommand initialGameCommand, List<String> initialInput) {
-        ChessBoard chessBoard = new ChessBoard();
-        GameCommand gameCommand = initialGameCommand;
-        List<String> input = initialInput;
-        while (GameCommand.isNotFinishedGame(gameCommand)) {
-            executeStartCommand(gameCommand, chessBoard);
-            executeMoveCommand(gameCommand, chessBoard, input);
-            input = repeatUntilSuccess(InputView::requestGameCommand);
-            gameCommand = GameCommand.findGameCommand(input.get(GAME_COMMAND_INDEX));
+    private void executeGameCommand(ChessBoard chessBoard) {
+        ChessGame chessGame = new ChessGame(chessBoard);
+        GameElements gameElements = requestGameCommand();
+        GameCommandState gameCommandState = gameElements.findGameCommandState();
+        while (!isEndState(gameElements)) {
```

Comment:
아래와 같은 모습이 더 메세지를 전달하는 모습이지 않을까요?
```suggestion
        while (gameElements.isNotEndState()) {
```